ampersand-state.js

```diff
@@ -177,11 +178,12 @@ assign(Base.prototype, Events, {
             }
 
             isEqual = this._getCompareForType(def.type);
+            onChange = this._getOnChangeForType(def.type);
             dataType = this._dataTypes[def.type];
 
             // check type if we have one
             if (dataType && dataType.set) {
-                cast = dataType.set(newVal);
+                cast = dataType.set.call(this, newVal, options);
```

Comment:
So here's at least one module using this in `change`: https://github.com/legastero/ampersand-wildemitter-datatype-mixin/blob/master/index.js. It makes me think maybe we shouldn't change that for now?

For one, it's kinda hard to deprecate, it's not easy to throw a warning if the user references this or something in the compare function. And secondly, datatype mixins like the above aren't generally pinned against a specific version of ampersand in anyway.